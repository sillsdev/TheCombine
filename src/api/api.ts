/* tslint:disable */
/* eslint-disable */
/**
 * BackendFramework
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

export * from "./api/audio-api";
export * from "./api/avatar-api";
export * from "./api/invite-api";
export * from "./api/lift-api";
export * from "./api/merge-api";
export * from "./api/project-api";
export * from "./api/user-api";
export * from "./api/user-edit-api";
export * from "./api/user-role-api";
export * from "./api/word-api";
