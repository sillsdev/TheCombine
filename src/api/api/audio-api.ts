/* tslint:disable */
/* eslint-disable */
/**
 * BackendFramework
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import globalAxios, { AxiosPromise, AxiosInstance } from "axios";
import { Configuration } from "../configuration";
// Some imports not used depending on template conditions
// @ts-ignore
import {
  DUMMY_BASE_URL,
  assertParamExists,
  setApiKeyToObject,
  setBasicAuthToObject,
  setBearerAuthToObject,
  setOAuthToObject,
  setSearchParams,
  serializeDataIfNeeded,
  toPathString,
  createRequestFunction,
} from "../common";
// @ts-ignore
import {
  BASE_PATH,
  COLLECTION_FORMATS,
  RequestArgs,
  BaseAPI,
  RequiredError,
} from "../base";
/**
 * AudioApi - axios parameter creator
 * @export
 */
export const AudioApiAxiosParamCreator = function (
  configuration?: Configuration
) {
  return {
    /**
     *
     * @param {string} projectId
     * @param {string} wordId
     * @param {string} fileName
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    deleteAudioFile: async (
      projectId: string,
      wordId: string,
      fileName: string,
      options: any = {}
    ): Promise<RequestArgs> => {
      // verify required parameter 'projectId' is not null or undefined
      assertParamExists("deleteAudioFile", "projectId", projectId);
      // verify required parameter 'wordId' is not null or undefined
      assertParamExists("deleteAudioFile", "wordId", wordId);
      // verify required parameter 'fileName' is not null or undefined
      assertParamExists("deleteAudioFile", "fileName", fileName);
      const localVarPath =
        `/v1/projects/{projectId}/words/{wordId}/audio/delete/{fileName}`
          .replace(`{${"projectId"}}`, encodeURIComponent(String(projectId)))
          .replace(`{${"wordId"}}`, encodeURIComponent(String(wordId)))
          .replace(`{${"fileName"}}`, encodeURIComponent(String(fileName)));
      // use dummy base URL string because the URL constructor only accepts absolute URLs.
      const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
      let baseOptions;
      if (configuration) {
        baseOptions = configuration.baseOptions;
      }

      const localVarRequestOptions = {
        method: "DELETE",
        ...baseOptions,
        ...options,
      };
      const localVarHeaderParameter = {} as any;
      const localVarQueryParameter = {} as any;

      setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
      let headersFromBaseOptions =
        baseOptions && baseOptions.headers ? baseOptions.headers : {};
      localVarRequestOptions.headers = {
        ...localVarHeaderParameter,
        ...headersFromBaseOptions,
        ...options.headers,
      };

      return {
        url: toPathString(localVarUrlObj),
        options: localVarRequestOptions,
      };
    },
    /**
     *
     * @param {string} projectId
     * @param {string} wordId
     * @param {string} fileName
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    downloadAudioFile: async (
      projectId: string,
      wordId: string,
      fileName: string,
      options: any = {}
    ): Promise<RequestArgs> => {
      // verify required parameter 'projectId' is not null or undefined
      assertParamExists("downloadAudioFile", "projectId", projectId);
      // verify required parameter 'wordId' is not null or undefined
      assertParamExists("downloadAudioFile", "wordId", wordId);
      // verify required parameter 'fileName' is not null or undefined
      assertParamExists("downloadAudioFile", "fileName", fileName);
      const localVarPath =
        `/v1/projects/{projectId}/words/{wordId}/audio/download/{fileName}`
          .replace(`{${"projectId"}}`, encodeURIComponent(String(projectId)))
          .replace(`{${"wordId"}}`, encodeURIComponent(String(wordId)))
          .replace(`{${"fileName"}}`, encodeURIComponent(String(fileName)));
      // use dummy base URL string because the URL constructor only accepts absolute URLs.
      const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
      let baseOptions;
      if (configuration) {
        baseOptions = configuration.baseOptions;
      }

      const localVarRequestOptions = {
        method: "GET",
        ...baseOptions,
        ...options,
      };
      const localVarHeaderParameter = {} as any;
      const localVarQueryParameter = {} as any;

      setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
      let headersFromBaseOptions =
        baseOptions && baseOptions.headers ? baseOptions.headers : {};
      localVarRequestOptions.headers = {
        ...localVarHeaderParameter,
        ...headersFromBaseOptions,
        ...options.headers,
      };

      return {
        url: toPathString(localVarUrlObj),
        options: localVarRequestOptions,
      };
    },
    /**
     *
     * @param {string} projectId
     * @param {string} wordId
     * @param {any} file
     * @param {string} name
     * @param {string} filePath
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    uploadAudioFile: async (
      projectId: string,
      wordId: string,
      file: any,
      name: string,
      filePath: string,
      options: any = {}
    ): Promise<RequestArgs> => {
      // verify required parameter 'projectId' is not null or undefined
      assertParamExists("uploadAudioFile", "projectId", projectId);
      // verify required parameter 'wordId' is not null or undefined
      assertParamExists("uploadAudioFile", "wordId", wordId);
      // verify required parameter 'file' is not null or undefined
      assertParamExists("uploadAudioFile", "file", file);
      // verify required parameter 'name' is not null or undefined
      assertParamExists("uploadAudioFile", "name", name);
      // verify required parameter 'filePath' is not null or undefined
      assertParamExists("uploadAudioFile", "filePath", filePath);
      const localVarPath =
        `/v1/projects/{projectId}/words/{wordId}/audio/upload`
          .replace(`{${"projectId"}}`, encodeURIComponent(String(projectId)))
          .replace(`{${"wordId"}}`, encodeURIComponent(String(wordId)));
      // use dummy base URL string because the URL constructor only accepts absolute URLs.
      const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
      let baseOptions;
      if (configuration) {
        baseOptions = configuration.baseOptions;
      }

      const localVarRequestOptions = {
        method: "POST",
        ...baseOptions,
        ...options,
      };
      const localVarHeaderParameter = {} as any;
      const localVarQueryParameter = {} as any;
      const localVarFormParams = new ((configuration &&
        configuration.formDataCtor) ||
        FormData)();

      if (file !== undefined) {
        localVarFormParams.append("File", file as any);
      }

      if (name !== undefined) {
        localVarFormParams.append("Name", name as any);
      }

      if (filePath !== undefined) {
        localVarFormParams.append("FilePath", filePath as any);
      }

      localVarHeaderParameter["Content-Type"] = "multipart/form-data";

      setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
      let headersFromBaseOptions =
        baseOptions && baseOptions.headers ? baseOptions.headers : {};
      localVarRequestOptions.headers = {
        ...localVarHeaderParameter,
        ...headersFromBaseOptions,
        ...options.headers,
      };
      localVarRequestOptions.data = localVarFormParams;

      return {
        url: toPathString(localVarUrlObj),
        options: localVarRequestOptions,
      };
    },
  };
};

/**
 * AudioApi - functional programming interface
 * @export
 */
export const AudioApiFp = function (configuration?: Configuration) {
  const localVarAxiosParamCreator = AudioApiAxiosParamCreator(configuration);
  return {
    /**
     *
     * @param {string} projectId
     * @param {string} wordId
     * @param {string} fileName
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    async deleteAudioFile(
      projectId: string,
      wordId: string,
      fileName: string,
      options?: any
    ): Promise<
      (axios?: AxiosInstance, basePath?: string) => AxiosPromise<string>
    > {
      const localVarAxiosArgs = await localVarAxiosParamCreator.deleteAudioFile(
        projectId,
        wordId,
        fileName,
        options
      );
      return createRequestFunction(
        localVarAxiosArgs,
        globalAxios,
        BASE_PATH,
        configuration
      );
    },
    /**
     *
     * @param {string} projectId
     * @param {string} wordId
     * @param {string} fileName
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    async downloadAudioFile(
      projectId: string,
      wordId: string,
      fileName: string,
      options?: any
    ): Promise<
      (axios?: AxiosInstance, basePath?: string) => AxiosPromise<any>
    > {
      const localVarAxiosArgs =
        await localVarAxiosParamCreator.downloadAudioFile(
          projectId,
          wordId,
          fileName,
          options
        );
      return createRequestFunction(
        localVarAxiosArgs,
        globalAxios,
        BASE_PATH,
        configuration
      );
    },
    /**
     *
     * @param {string} projectId
     * @param {string} wordId
     * @param {any} file
     * @param {string} name
     * @param {string} filePath
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    async uploadAudioFile(
      projectId: string,
      wordId: string,
      file: any,
      name: string,
      filePath: string,
      options?: any
    ): Promise<
      (axios?: AxiosInstance, basePath?: string) => AxiosPromise<string>
    > {
      const localVarAxiosArgs = await localVarAxiosParamCreator.uploadAudioFile(
        projectId,
        wordId,
        file,
        name,
        filePath,
        options
      );
      return createRequestFunction(
        localVarAxiosArgs,
        globalAxios,
        BASE_PATH,
        configuration
      );
    },
  };
};

/**
 * AudioApi - factory interface
 * @export
 */
export const AudioApiFactory = function (
  configuration?: Configuration,
  basePath?: string,
  axios?: AxiosInstance
) {
  const localVarFp = AudioApiFp(configuration);
  return {
    /**
     *
     * @param {string} projectId
     * @param {string} wordId
     * @param {string} fileName
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    deleteAudioFile(
      projectId: string,
      wordId: string,
      fileName: string,
      options?: any
    ): AxiosPromise<string> {
      return localVarFp
        .deleteAudioFile(projectId, wordId, fileName, options)
        .then((request) => request(axios, basePath));
    },
    /**
     *
     * @param {string} projectId
     * @param {string} wordId
     * @param {string} fileName
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    downloadAudioFile(
      projectId: string,
      wordId: string,
      fileName: string,
      options?: any
    ): AxiosPromise<any> {
      return localVarFp
        .downloadAudioFile(projectId, wordId, fileName, options)
        .then((request) => request(axios, basePath));
    },
    /**
     *
     * @param {string} projectId
     * @param {string} wordId
     * @param {any} file
     * @param {string} name
     * @param {string} filePath
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    uploadAudioFile(
      projectId: string,
      wordId: string,
      file: any,
      name: string,
      filePath: string,
      options?: any
    ): AxiosPromise<string> {
      return localVarFp
        .uploadAudioFile(projectId, wordId, file, name, filePath, options)
        .then((request) => request(axios, basePath));
    },
  };
};

/**
 * Request parameters for deleteAudioFile operation in AudioApi.
 * @export
 * @interface AudioApiDeleteAudioFileRequest
 */
export interface AudioApiDeleteAudioFileRequest {
  /**
   *
   * @type {string}
   * @memberof AudioApiDeleteAudioFile
   */
  readonly projectId: string;

  /**
   *
   * @type {string}
   * @memberof AudioApiDeleteAudioFile
   */
  readonly wordId: string;

  /**
   *
   * @type {string}
   * @memberof AudioApiDeleteAudioFile
   */
  readonly fileName: string;
}

/**
 * Request parameters for downloadAudioFile operation in AudioApi.
 * @export
 * @interface AudioApiDownloadAudioFileRequest
 */
export interface AudioApiDownloadAudioFileRequest {
  /**
   *
   * @type {string}
   * @memberof AudioApiDownloadAudioFile
   */
  readonly projectId: string;

  /**
   *
   * @type {string}
   * @memberof AudioApiDownloadAudioFile
   */
  readonly wordId: string;

  /**
   *
   * @type {string}
   * @memberof AudioApiDownloadAudioFile
   */
  readonly fileName: string;
}

/**
 * Request parameters for uploadAudioFile operation in AudioApi.
 * @export
 * @interface AudioApiUploadAudioFileRequest
 */
export interface AudioApiUploadAudioFileRequest {
  /**
   *
   * @type {string}
   * @memberof AudioApiUploadAudioFile
   */
  readonly projectId: string;

  /**
   *
   * @type {string}
   * @memberof AudioApiUploadAudioFile
   */
  readonly wordId: string;

  /**
   *
   * @type {any}
   * @memberof AudioApiUploadAudioFile
   */
  readonly file: any;

  /**
   *
   * @type {string}
   * @memberof AudioApiUploadAudioFile
   */
  readonly name: string;

  /**
   *
   * @type {string}
   * @memberof AudioApiUploadAudioFile
   */
  readonly filePath: string;
}

/**
 * AudioApi - object-oriented interface
 * @export
 * @class AudioApi
 * @extends {BaseAPI}
 */
export class AudioApi extends BaseAPI {
  /**
   *
   * @param {AudioApiDeleteAudioFileRequest} requestParameters Request parameters.
   * @param {*} [options] Override http request option.
   * @throws {RequiredError}
   * @memberof AudioApi
   */
  public deleteAudioFile(
    requestParameters: AudioApiDeleteAudioFileRequest,
    options?: any
  ) {
    return AudioApiFp(this.configuration)
      .deleteAudioFile(
        requestParameters.projectId,
        requestParameters.wordId,
        requestParameters.fileName,
        options
      )
      .then((request) => request(this.axios, this.basePath));
  }

  /**
   *
   * @param {AudioApiDownloadAudioFileRequest} requestParameters Request parameters.
   * @param {*} [options] Override http request option.
   * @throws {RequiredError}
   * @memberof AudioApi
   */
  public downloadAudioFile(
    requestParameters: AudioApiDownloadAudioFileRequest,
    options?: any
  ) {
    return AudioApiFp(this.configuration)
      .downloadAudioFile(
        requestParameters.projectId,
        requestParameters.wordId,
        requestParameters.fileName,
        options
      )
      .then((request) => request(this.axios, this.basePath));
  }

  /**
   *
   * @param {AudioApiUploadAudioFileRequest} requestParameters Request parameters.
   * @param {*} [options] Override http request option.
   * @throws {RequiredError}
   * @memberof AudioApi
   */
  public uploadAudioFile(
    requestParameters: AudioApiUploadAudioFileRequest,
    options?: any
  ) {
    return AudioApiFp(this.configuration)
      .uploadAudioFile(
        requestParameters.projectId,
        requestParameters.wordId,
        requestParameters.file,
        requestParameters.name,
        requestParameters.filePath,
        options
      )
      .then((request) => request(this.axios, this.basePath));
  }
}
