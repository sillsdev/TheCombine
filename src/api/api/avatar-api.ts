/* tslint:disable */
/* eslint-disable */
/**
 * BackendFramework
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import globalAxios, { AxiosPromise, AxiosInstance } from "axios";
import { Configuration } from "../configuration";
// Some imports not used depending on template conditions
// @ts-ignore
import {
  DUMMY_BASE_URL,
  assertParamExists,
  setApiKeyToObject,
  setBasicAuthToObject,
  setBearerAuthToObject,
  setOAuthToObject,
  setSearchParams,
  serializeDataIfNeeded,
  toPathString,
  createRequestFunction,
} from "../common";
// @ts-ignore
import {
  BASE_PATH,
  COLLECTION_FORMATS,
  RequestArgs,
  BaseAPI,
  RequiredError,
} from "../base";
/**
 * AvatarApi - axios parameter creator
 * @export
 */
export const AvatarApiAxiosParamCreator = function (
  configuration?: Configuration
) {
  return {
    /**
     *
     * @param {string} userId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    downloadAvatar: async (
      userId: string,
      options: any = {}
    ): Promise<RequestArgs> => {
      // verify required parameter 'userId' is not null or undefined
      assertParamExists("downloadAvatar", "userId", userId);
      const localVarPath = `/v1/users/{userId}/avatar/download`.replace(
        `{${"userId"}}`,
        encodeURIComponent(String(userId))
      );
      // use dummy base URL string because the URL constructor only accepts absolute URLs.
      const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
      let baseOptions;
      if (configuration) {
        baseOptions = configuration.baseOptions;
      }

      const localVarRequestOptions = {
        method: "GET",
        ...baseOptions,
        ...options,
      };
      const localVarHeaderParameter = {} as any;
      const localVarQueryParameter = {} as any;

      setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
      let headersFromBaseOptions =
        baseOptions && baseOptions.headers ? baseOptions.headers : {};
      localVarRequestOptions.headers = {
        ...localVarHeaderParameter,
        ...headersFromBaseOptions,
        ...options.headers,
      };

      return {
        url: toPathString(localVarUrlObj),
        options: localVarRequestOptions,
      };
    },
    /**
     *
     * @param {string} userId
     * @param {any} file
     * @param {string} name
     * @param {string} filePath
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    uploadAvatar: async (
      userId: string,
      file: any,
      name: string,
      filePath: string,
      options: any = {}
    ): Promise<RequestArgs> => {
      // verify required parameter 'userId' is not null or undefined
      assertParamExists("uploadAvatar", "userId", userId);
      // verify required parameter 'file' is not null or undefined
      assertParamExists("uploadAvatar", "file", file);
      // verify required parameter 'name' is not null or undefined
      assertParamExists("uploadAvatar", "name", name);
      // verify required parameter 'filePath' is not null or undefined
      assertParamExists("uploadAvatar", "filePath", filePath);
      const localVarPath = `/v1/users/{userId}/avatar/upload`.replace(
        `{${"userId"}}`,
        encodeURIComponent(String(userId))
      );
      // use dummy base URL string because the URL constructor only accepts absolute URLs.
      const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
      let baseOptions;
      if (configuration) {
        baseOptions = configuration.baseOptions;
      }

      const localVarRequestOptions = {
        method: "POST",
        ...baseOptions,
        ...options,
      };
      const localVarHeaderParameter = {} as any;
      const localVarQueryParameter = {} as any;
      const localVarFormParams = new ((configuration &&
        configuration.formDataCtor) ||
        FormData)();

      if (file !== undefined) {
        localVarFormParams.append("File", file as any);
      }

      if (name !== undefined) {
        localVarFormParams.append("Name", name as any);
      }

      if (filePath !== undefined) {
        localVarFormParams.append("FilePath", filePath as any);
      }

      localVarHeaderParameter["Content-Type"] = "multipart/form-data";

      setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
      let headersFromBaseOptions =
        baseOptions && baseOptions.headers ? baseOptions.headers : {};
      localVarRequestOptions.headers = {
        ...localVarHeaderParameter,
        ...headersFromBaseOptions,
        ...options.headers,
      };
      localVarRequestOptions.data = localVarFormParams;

      return {
        url: toPathString(localVarUrlObj),
        options: localVarRequestOptions,
      };
    },
  };
};

/**
 * AvatarApi - functional programming interface
 * @export
 */
export const AvatarApiFp = function (configuration?: Configuration) {
  const localVarAxiosParamCreator = AvatarApiAxiosParamCreator(configuration);
  return {
    /**
     *
     * @param {string} userId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    async downloadAvatar(
      userId: string,
      options?: any
    ): Promise<
      (axios?: AxiosInstance, basePath?: string) => AxiosPromise<any>
    > {
      const localVarAxiosArgs = await localVarAxiosParamCreator.downloadAvatar(
        userId,
        options
      );
      return createRequestFunction(
        localVarAxiosArgs,
        globalAxios,
        BASE_PATH,
        configuration
      );
    },
    /**
     *
     * @param {string} userId
     * @param {any} file
     * @param {string} name
     * @param {string} filePath
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    async uploadAvatar(
      userId: string,
      file: any,
      name: string,
      filePath: string,
      options?: any
    ): Promise<
      (axios?: AxiosInstance, basePath?: string) => AxiosPromise<void>
    > {
      const localVarAxiosArgs = await localVarAxiosParamCreator.uploadAvatar(
        userId,
        file,
        name,
        filePath,
        options
      );
      return createRequestFunction(
        localVarAxiosArgs,
        globalAxios,
        BASE_PATH,
        configuration
      );
    },
  };
};

/**
 * AvatarApi - factory interface
 * @export
 */
export const AvatarApiFactory = function (
  configuration?: Configuration,
  basePath?: string,
  axios?: AxiosInstance
) {
  const localVarFp = AvatarApiFp(configuration);
  return {
    /**
     *
     * @param {string} userId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    downloadAvatar(userId: string, options?: any): AxiosPromise<any> {
      return localVarFp
        .downloadAvatar(userId, options)
        .then((request) => request(axios, basePath));
    },
    /**
     *
     * @param {string} userId
     * @param {any} file
     * @param {string} name
     * @param {string} filePath
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    uploadAvatar(
      userId: string,
      file: any,
      name: string,
      filePath: string,
      options?: any
    ): AxiosPromise<void> {
      return localVarFp
        .uploadAvatar(userId, file, name, filePath, options)
        .then((request) => request(axios, basePath));
    },
  };
};

/**
 * Request parameters for downloadAvatar operation in AvatarApi.
 * @export
 * @interface AvatarApiDownloadAvatarRequest
 */
export interface AvatarApiDownloadAvatarRequest {
  /**
   *
   * @type {string}
   * @memberof AvatarApiDownloadAvatar
   */
  readonly userId: string;
}

/**
 * Request parameters for uploadAvatar operation in AvatarApi.
 * @export
 * @interface AvatarApiUploadAvatarRequest
 */
export interface AvatarApiUploadAvatarRequest {
  /**
   *
   * @type {string}
   * @memberof AvatarApiUploadAvatar
   */
  readonly userId: string;

  /**
   *
   * @type {any}
   * @memberof AvatarApiUploadAvatar
   */
  readonly file: any;

  /**
   *
   * @type {string}
   * @memberof AvatarApiUploadAvatar
   */
  readonly name: string;

  /**
   *
   * @type {string}
   * @memberof AvatarApiUploadAvatar
   */
  readonly filePath: string;
}

/**
 * AvatarApi - object-oriented interface
 * @export
 * @class AvatarApi
 * @extends {BaseAPI}
 */
export class AvatarApi extends BaseAPI {
  /**
   *
   * @param {AvatarApiDownloadAvatarRequest} requestParameters Request parameters.
   * @param {*} [options] Override http request option.
   * @throws {RequiredError}
   * @memberof AvatarApi
   */
  public downloadAvatar(
    requestParameters: AvatarApiDownloadAvatarRequest,
    options?: any
  ) {
    return AvatarApiFp(this.configuration)
      .downloadAvatar(requestParameters.userId, options)
      .then((request) => request(this.axios, this.basePath));
  }

  /**
   *
   * @param {AvatarApiUploadAvatarRequest} requestParameters Request parameters.
   * @param {*} [options] Override http request option.
   * @throws {RequiredError}
   * @memberof AvatarApi
   */
  public uploadAvatar(
    requestParameters: AvatarApiUploadAvatarRequest,
    options?: any
  ) {
    return AvatarApiFp(this.configuration)
      .uploadAvatar(
        requestParameters.userId,
        requestParameters.file,
        requestParameters.name,
        requestParameters.filePath,
        options
      )
      .then((request) => request(this.axios, this.basePath));
  }
}
