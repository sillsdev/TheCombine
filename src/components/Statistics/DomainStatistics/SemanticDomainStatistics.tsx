import { Card, Grid, ListItem, List } from "@mui/material";
import { ReactElement, useState, useEffect } from "react";

import { SemanticDomainCount, SemanticDomainTreeNode } from "api/models";
import { getSemanticDomainCounts } from "backend";
import * as LocalStorage from "backend/localStorage";
import { ColumnHead, TableCell } from "components/Statistics/TableCells";

interface SemanticDomainStatisticsProps {
  lang: string;
}

export default function SemanticDomainStatistics(
  props: SemanticDomainStatisticsProps
): ReactElement {
  const [statisticsList, setStatisticsList] = useState<SemanticDomainCount[]>(
    []
  );

  useEffect(() => {
    const updateStatisticList = async () => {
      const counts = await getStatisticsCounts(
        LocalStorage.getProjectId(),
        props.lang
      );
      if (counts !== undefined) {
        return setStatisticsList(counts);
      }
    };
    updateStatisticList();
  }, [props.lang]);

  async function getStatisticsCounts(
    projectId: string,
    lang?: string
  ): Promise<SemanticDomainCount[] | undefined> {
    return await getSemanticDomainCounts(projectId, lang);
  }

  return (
    <Grid container justifyContent="center">
      <Card style={{ width: 600 }}>
        <List>
          <Grid container wrap="nowrap" justifyContent="space-around">
            <ColumnHead titleId={"statistics.domainNumber"} />
            <ColumnHead titleId={"statistics.domainName"} />
            <ColumnHead titleId={"statistics.countSenses"} />
          </Grid>
        </List>
        <List>
          {statisticsList.map((t) => (
            <TableRow
              key={t.semanticDomainTreeNode.id}
              dom={t.semanticDomainTreeNode}
              count={t.count}
            />
          ))}
        </List>
      </Card>
    </Grid>
  );
}

function TableRow(props: {
  dom: SemanticDomainTreeNode;
  count: number;
}): ReactElement {
  return (
    <ListItem style={{ minWidth: "600px" }}>
      <Grid container wrap="nowrap" justifyContent="space-around">
        <TableCell key={"id_" + props.dom.id} text={props.dom.id} />
        <TableCell key={"domain_" + props.dom.id} text={props.dom.name} />
        <TableCell key={"count_" + props.dom.id} text={props.count} />
      </Grid>
    </ListItem>
  );
}
