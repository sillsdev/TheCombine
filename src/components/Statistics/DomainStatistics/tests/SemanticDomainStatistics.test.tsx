import { ListItem } from "@material-ui/core";
import renderer, { ReactTestRenderer } from "react-test-renderer";

import "tests/mockReactI18next";

import { SemanticDomainCount } from "api";
import SemanticDomainStatistics from "components/Statistics/DomainStatistics/SemanticDomainStatistics";
import {
  newSemanticDomainCount,
  newSemanticDomainTreeNode,
} from "types/semanticDomain";

let testRenderer: ReactTestRenderer;

const mockProjectId = "mockProjectId";
const mockTreeNode = newSemanticDomainTreeNode();
const mockSemanticDomainCountArray: Array<SemanticDomainCount> = [
  newSemanticDomainCount(mockTreeNode),
];

const mockGetStatisticsCounts = jest.fn();
const mockGetProjectId = jest.fn();

jest.mock("backend", () => ({
  getSemanticDomainCounts: (projectId: string, lang?: string) =>
    mockGetStatisticsCounts(projectId, lang),
}));

jest.mock("backend/localStorage", () => ({
  getProjectId: () => mockGetProjectId(),
}));

function setMockFunctions() {
  mockGetProjectId.mockReturnValue(mockProjectId);
  mockGetStatisticsCounts.mockResolvedValue(mockSemanticDomainCountArray);
}

beforeEach(async () => {
  jest.clearAllMocks();
  setMockFunctions();
  await renderer.act(async () => {
    testRenderer = renderer.create(<SemanticDomainStatistics lang={""} />);
  });
});

describe("SemanticDomainStatistics", () => {
  it("renders without crashing, UI does not change unexpectedly", async () => {
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it("useEffect hook was called", async () => {
    //Verify the mock function called
    expect(mockGetProjectId).toBeCalled();

    //Verify ListItem for the SemanticDomainCount object is present
    const newSenDomCountList = testRenderer.root.findAllByType(ListItem);
    expect(newSenDomCountList.length).toEqual(
      mockSemanticDomainCountArray.length
    );
  });
});
