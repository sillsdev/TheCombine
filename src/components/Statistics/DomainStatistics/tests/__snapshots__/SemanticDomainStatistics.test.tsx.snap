// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SemanticDomainStatistics renders without crashing, UI does not change unexpectedly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    style={
      Object {
        "width": 600,
      }
    }
  >
    <ul
      className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-1ymxtd0-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
          style={
            Object {
              "borderBottomStyle": "dotted",
              "borderBottomWidth": 1,
              "position": "relative",
            }
          }
        >
          <h6
            className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
          >
            statistics.domainNumber
          </h6>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
          style={
            Object {
              "borderBottomStyle": "dotted",
              "borderBottomWidth": 1,
              "position": "relative",
            }
          }
        >
          <h6
            className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
          >
            statistics.domainName
          </h6>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
          style={
            Object {
              "borderBottomStyle": "dotted",
              "borderBottomWidth": 1,
              "position": "relative",
            }
          }
        >
          <h6
            className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
          >
            statistics.countSenses
          </h6>
        </div>
      </div>
    </ul>
    <ul
      className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
        disabled={false}
        style={
          Object {
            "minWidth": "600px",
          }
        }
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-1ymxtd0-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
            style={
              Object {
                "borderBottomStyle": "dotted",
                "borderBottomWidth": 1,
                "position": "relative",
              }
            }
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              
            </p>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
            style={
              Object {
                "borderBottomStyle": "dotted",
                "borderBottomWidth": 1,
                "position": "relative",
              }
            }
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              
            </p>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
            style={
              Object {
                "borderBottomStyle": "dotted",
                "borderBottomWidth": 1,
                "position": "relative",
              }
            }
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              0
            </p>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
